build.sh

```diff
@@ -274,78 +290,91 @@ case "$COMMAND" in
     "test")
         set +e # Run both sets of tests even if the first fails
         failed=0
-        sh build.sh test-ios "$XCMODE" || failed=1
-        sh build.sh test-ios-devices "$XCMODE" || failed=1
-        sh build.sh test-osx "$XCMODE" || failed=1
-        exit $failed
-        ;;
-
-    "test-debug")
-        set +e
-        failed=0
-        sh build.sh test-ios-debug "$XCMODE" || failed=1
-        sh build.sh test-ios-devices-debug "$XCMODE" || failed=1
-        sh build.sh test-osx-debug "$XCMODE" || failed=1
+        sh build.sh test-ios || failed=1
+        sh build.sh test-ios-devices || failed=1
+        sh build.sh test-osx || failed=1
         exit $failed
         ;;
 
     "test-all")
         set +e
         failed=0
-        sh build.sh test "$XCMODE" || failed=1
-        sh build.sh test-debug "$XCMODE" || failed=1
+        sh build.sh test || failed=1
+        sh build.sh test-debug || failed=1
         exit $failed
         ;;
 
     "test-ios")
-        xcrealm "-scheme iOS -configuration Release -sdk iphonesimulator -destination 'name=iPhone 6' test"
-        xcrealm "-scheme iOS -configuration Release -sdk iphonesimulator -destination 'name=iPhone 4S' test"
-        xcrealm "-scheme 'iOS 8' -configuration Release -sdk iphonesimulator -destination 'name=iPhone 6' test"
+        xcrealm "-scheme iOS -configuration $CONFIGURATION -sdk iphonesimulator -destination 'name=iPhone 6' test"
+        xcrealm "-scheme iOS -configuration $CONFIGURATION -sdk iphonesimulator -destination 'name=iPhone 4S' test"
+        xcrealm "-scheme 'iOS 8' -configuration $CONFIGURATION -sdk iphonesimulator -destination 'name=iPhone 6' test"
         exit 0
         ;;
 
     "test-ios-devices")
-        test_ios_devices "Release" "$XCMODE"
+        test_ios_devices "$CONFIGURATION"
         ;;
 
     "test-osx")
-        xcrealm "-scheme OSX -configuration Release test"
+        xcrealm "-scheme OSX -configuration $CONFIGURATION test"
         exit 0
         ;;
 
-    "test-ios-debug")
-        xcrealm "-scheme iOS -configuration Debug -sdk iphonesimulator -destination 'name=iPhone 6' test"
-        xcrealm "-scheme iOS -configuration Debug -sdk iphonesimulator -destination 'name=iPhone 4S' test"
-        xcrealm "-scheme 'iOS 8' -configuration Debug -sdk iphonesimulator -destination 'name=iPhone 6' test"
+    ######################################
+    # Full verification
+    ######################################
+    "verify")
+        sh build.sh verify-docs
+        sh build.sh verify-osx
+        sh build.sh verify-osx-debug
+        sh build.sh verify-ios
+        sh build.sh verify-ios-debug
+        sh build.sh verify-ios-device
+        ;;
+
+    "verify-osx")
+        CONFIGURATION="$CONFIGURATION" sh build.sh test-osx
```

Comment:
Setting `CONFIGURATION` to its existing value actually does something? If so, we should do it everywhere else we call `sh build.sh`.